internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -940,11 +945,11 @@ func getUpgradeRequest(d *schema.ResourceData) *admin.LegacyAtlasTenantClusterUp
 
 func updateAdvancedCluster(
 	ctx context.Context,
-	connV2 *admin.APIClient,
-	request *admin.AdvancedClusterDescription,
+	connV2 *admin20231115.APIClient,
+	request *admin20231115.AdvancedClusterDescription,
 	projectID, name string,
 	timeout time.Duration,
-) (*admin.AdvancedClusterDescription, *http.Response, error) {
+) (*admin20231115.AdvancedClusterDescription, *http.Response, error) {
```

Comment:
will tests go in a different PR?

Reply:
yes, to define a working acceptance test with symmetric sharded cluster and asymmetric sharded cluster using the new schema first the read operation needs to be adjusted to call the new API (coming in separate PR).